h2o-core/src/main/java/water/MRTask.java

```diff
@@ -378,36 +379,115 @@ private StringBuilder print(StringBuilder sb, int d) {
+    return doAll(null, fr, false); 
+  }
+
+  /** 
+   * Invokes the map/reduce computation over the given Frame. This call is
+   * blocking. The run is performed across the cluster.
+   *  
+   * @param outputTypes The type of output Vec instances to create. See {@link Vec#T_STR}, {@link Vec#T_NUM}, 
+   *    {@link Vec#T_CAT} and other byte constatnts in {@link Vec} to see possible values.
+   *
+   * @param fr Perform the computation on this Frame instance. This frame can be possibly mutated as side-effect.   
+   *
+   * @return this
+   */
+  public final T doAll(byte[] outputTypes, Frame fr) {
+    return doAll(outputTypes, fr, false);
+  }
+
+  /** 
+   * Invokes the map/reduce computation over the given Frame.  This call is
+   * blocking. The run is performed across the cluster.
+   *  
+   * @param outputType The type of one output Vec instance to create. See {@link Vec#T_STR}, {@link Vec#T_NUM}, 
+   *    {@link Vec#T_CAT} and other byte constatnts in {@link Vec} to see possible values.
+   *
+   * @param fr  Perform the computation on this Frame instance. This frame can be possibly mutated as side-effect. 
+   *
+   * @return this
+   */
+  public final T doAll(byte outputType, Frame fr) {
+    return doAll(new byte[]{outputType}, fr, false);
+  }
+
+  /** 
+   * Invokes the map/reduce computation over the given Frame.  This call is
+   * blocking. 
+   *
+   * @param outputTypes The type of output Vec instances to create. See {@link Vec#T_STR}, {@link Vec#T_NUM}, 
+   *    {@link Vec#T_CAT} and other byte constatnts in {@link Vec} to see possible values.
+   *
+   * @param fr Perform the computation on this Frame instance. This frame can be possibly mutated as side-effect. 
+   *
+   * @param runLocal Run locally by copying data, or run across the cluster?
+   *
+   * @return this
+   */
+  public final T doAll(byte[] outputTypes, Frame fr, boolean runLocal) {
+    dfork(outputTypes,fr, runLocal);
     return getResult();
   }
-  // Output is several vecs of the same type
-  public final T doAll( int nouts, byte type, Frame fr) {
-    byte[] types = new byte[nouts];
-    Arrays.fill(types, type);
-    return doAll(types,fr,false);
+
+  /**
+   * Invokes the map/reduce computation over the given Frame.  This call is
+   * blocking. 
+   * 
+   * @param numverOfOutputs Number of output vectors for the computation. All of them will have the same type.
+   *
+   * @param outputType The type of all the output Vec instances to create. See {@link Vec#T_STR}, {@link Vec#T_NUM}, 
+   *    {@link Vec#T_CAT} and other byte constatnts in {@link Vec} to see possible values.
+   *
+   * @param fr Perform the computation on this Frame instance. This frame can be possibly mutated as side-effect.   
+   *
+   * @return this
+   */ 
+  public final T doAll(int numverOfOutputs, byte outputType, Frame fr) {
+    byte[] types = new byte[numverOfOutputs];
+    Arrays.fill(types, outputType);
+    return doAll(types, fr,false);
```

Comment:
missing space between fr and false (when it was added between types and fr)

Reply:
fixed